### PR TITLE
[DOCS] Separate Enrich API Docs

### PR DESCRIPTION
**Pull request by jrodewig**
### Changes
- Moves enrich API documentation into separate files in the [REST APIs](https://www.elastic.co/guide/en/elasticsearch/reference/master/rest-apis.html) section.
- Reformats enrich API docs to follow the [API Reference Docs template](https://github.com/elastic/docs/blame/master/shared/api-ref-ex.asciidoc)

### Dependencies
- #46041 (Could be removed if needed)

### CODE REVIEW

File: docs/reference/ingest/apis/enrich/delete-enrich-policy.asciidoc [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+==== {api-description-title}
+
+Use the delete enrich policy API
+to delete an existing enrich policy
+and its enrich index.
+
+[IMPORTANT]
+====
+You must remove an enrich policy
+from any in-use enrich processors and ingest pipelines
```
Comment (martijnvg) [inline docs/reference/ingest/apis/enrich/delete-enrich-policy.asciidoc]: maybe change `enrich processors and ingest pipelines` into `ingest pipeline`?
Processors are part of a pipeline's configuration, so the latter is less redundant.

File: docs/reference/ingest/apis/enrich/put-enrich-policy.asciidoc [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+// end::enrich-policy-def
+
+
+===== Update an enrich policy
+
+// tag::update-enrich-policy
+You cannot update an existing enrich policy.
+Instead, you can:
+
+.   Create a new enrich policy
```
Comment (martijnvg) [inline docs/reference/ingest/apis/enrich/put-enrich-policy.asciidoc]: Maybe change `Create a new enrich policy` into `Create and execute a new enrich policy`?
The new policy cannot be configured in a pipeline if it hasn't been executed before.

File: docs/reference/ingest/apis/enrich/put-enrich-policy.asciidoc [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+    containing your desired changes.
+
+.   Remove the previous enrich policy
+    from any in-use enrich processors
+    and ingest pipelines.
+
+.   Add the new enrich policy
```
Comment (martijnvg) [inline docs/reference/ingest/apis/enrich/put-enrich-policy.asciidoc]: Maybe combine these two points into one? And phrase it as replace the policy name with the name of the new enrich policy

**Review by martijnvg — COMMENTED** [review]
Looks good! I left a few small comments. 
Also I will merge #46041, so that the docs build doesn't fail, because `match` policy type does not exist.

File: docs/reference/ingest/apis/enrich/delete-enrich-policy.asciidoc (hunk 1 above)
Comment (jrodewig) [reply docs/reference/ingest/apis/enrich/delete-enrich-policy.asciidoc]: Fixed with 0f17b487f90fba0677aba9f4240ecdb8f33d7777.

File: docs/reference/ingest/apis/enrich/put-enrich-policy.asciidoc (hunk 2 above)
Comment (jrodewig) [reply docs/reference/ingest/apis/enrich/put-enrich-policy.asciidoc]: Fixed with 928f62e329aff270e78a986f2ad0a733cef1f6c4.

File: docs/reference/ingest/apis/enrich/put-enrich-policy.asciidoc (hunk 3 above)
Comment (jrodewig) [reply docs/reference/ingest/apis/enrich/put-enrich-policy.asciidoc]: Fixed with 928f62e329aff270e78a986f2ad0a733cef1f6c4.

**Comment by jrodewig** [comment]
Thanks @martijnvg!

**Review by martijnvg — APPROVED** [review]
LGTM (assuming docs build succeeds)

The fact that in the old api description the delete policy docs was the last, helped with cleaning up the enrich policies :) So I guess some pages will need a hidden delete policy api invocation.

**Pull request merged**